lib/wos_records.rb

```diff
@@ -0,0 +1,128 @@
+require 'forwardable'
+require 'htmlentities'
+
+# Utilities for working with Web of Science records
+class WosRecords
+  extend Forwardable
+  include Enumerable
+
+  # @return [Integer] WOS record count
+  def_delegators :rec_nodes, :count
+
+  # @return xml [String] WOS records in XML
+  def_delegators :doc, :to_xml
+
+  # @return doc [Nokogiri::XML::Document] WOS records document
+  attr_reader :doc
+
+  # @param records [String] records in XML
+  # @param encoded_records [String] records in HTML encoding
+  def initialize(records: nil, encoded_records: nil)
+    @records = records
+    @encoded_records = encoded_records
+    @records = decode_records if records.nil? && !encoded_records.nil?
+    @doc = Nokogiri::XML(@records) { |config| config.strict.noblanks }
+  end
+
+  # Iterate over the REC nodes
+  # @yield rec [Nokogiri::XML::Element]
+  def each
+    rec_nodes.each { |rec| yield rec }
+  end
+
+  # @return uids [Array<String>] the rec_nodes UID values (in order)
+  def uids
+    uid_nodes.map(&:text)
+  end
+
+  # @return uid_nodes [Nokogiri::XML::NodeSet] the rec_nodes UID nodes
+  def uid_nodes
+    rec_nodes.search('UID')
+  end
+
+  # Find duplicate WoS UID values
+  # @param record_setB [WosRecords]
+  # @return [Set] duplicate WoS UID values
+  def duplicate_uids(record_setB)
+    uids.to_set.intersection(record_setB.uids.to_set)
```

Comment:
`Array` already has an [intersection operator](https://ruby-doc.org/core-2.2.0/Array.html#method-i-26) in Ruby, the `&`.  So without the need to convert `to_set`, this should be `uids & record_setB.uids`.  Intersection also deduplicates multiples and preserves order!

Reply:
OK, I'll try that, good tip.

Reply:
Yep, this works for me:
```ruby
> x = [*1..10]
=> [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
> y = [*1..10].reject {|i| i % 2 == 0 }.reverse
=> [9, 7, 5, 3, 1]
> x & y
=> [1, 3, 5, 7, 9]
> x.reverse & y
=> [9, 7, 5, 3, 1]
```

Reply:
Done